### PR TITLE
Id.Stream to store Id.Namespace instead of String

### PR DESCRIPTION
**Pull request by anwar6953**
https://issues.cask.co/browse/CDAP-1486
http://builds.cask.co/browse/CDAP-DUT1003-2

### CODE REVIEW

**Comment by bdmogal** [comment]
LGTM pending build.

File: cdap-proto/src/main/java/co/cask/cdap/proto/Id.java [hunk 1]
```diff
@@ -383,18 +383,17 @@ public int hashCode() {
    * Id.Stream uniquely identifies a stream.
    */
   public static final class Stream {
-    private final String namespace;
+    private final Namespace namespace;
```
Comment (chtyim) [inline cdap-proto/src/main/java/co/cask/cdap/proto/Id.java]: this affect the efficiency of the computation of hashCode for `Id.Stream`. You'd like to change the `Id.Namespace` hashCode computation to be efficient as well.

Comment (anwar6953) [inline cdap-proto/src/main/java/co/cask/cdap/proto/Id.java]: We cache the hashcode for StreamId, so the hashcode will only be computed once for the object.

That should be fine, right

Comment (chtyim) [inline cdap-proto/src/main/java/co/cask/cdap/proto/Id.java]: no. The computation itself need to be fast, as you still computing it on every event (from the StreamHandler, you construct a new Id.Stream object and that will be used to lookup a Map, and that's when the hashCode is computed on the newly created Id.Stream object. The cache only saves the one in the Map already).

Comment (anwar6953) [inline cdap-proto/src/main/java/co/cask/cdap/proto/Id.java]: ok, made Namespace's hashcode just the hashcode of the string it consists of.

**Comment by chtyim** [comment]
LGTM

**Pull request merged**